debug-tools/snap-debug-info.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+
+h1(){ echo -e "\n==================== $* ===================="; }
+h2(){ echo -e "\n========== $* =========="; }
+h3(){ echo -e "\n===== $* ====="; }
+
+h1 "SNAP VERSION"; snap version
+h1 "SNAP WHOAMI"; snap whoami
+h1 "SNAP MODEL"; snap model --verbose
+h1 "SNAP MODEL SERIAL"; snap model --serial --verbose
+h1 "SNAP LIST"; snap list
+h1 "SNAP SERVICES"; snap services
+h1 "SNAP CONNECTIONS"; snap connections
+
+h1 "PER-SNAP CONNECTIONS"
+for sn in $(snap list | awk 'NR>1 {print $1}'); do
+    h2 "PER-SNAP $sn CONNECTIONS"
+    snap connections "$sn"
+done
+h1 "SNAP CHANGES"
+snap changes --abs-time
+h1 "GADGET SNAP GADGET.YAML"
+cat /snap/"$(snap list | awk '($6 ~ /.*gadget.*$/) {print $1}')"/current/meta/gadget.yaml
+
+h1 "SNAP CHANGES (in Doing)"
+# print off the output of snap tasks <chg> for every chg that is in Doing state
+for chg in $(snap changes | tail -n +2 | grep -Po '(?:[0-9]+\s+Doing)' | awk '{print $1}'); do
+    h3 "tasks for $chg"
+    snap tasks "$chg" --abs-time
+done
+
+h1 "SNAP CHANGES (in Error)"
+# same as above, just for Error instead of Doing
+for chg in $(snap changes | tail -n +2 | grep -Po '(?:[0-9]+\s+Error)' | awk '{print $1}'); do
+    h3 "tasks for $chg"
+    snap tasks "$chg" --abs-time
+done
+
+exit 0
```

Comment:
+1, and also:

1. some of the commands below do not require sudo, se can move them up
2. Let's remove `sudo` from the commands below (since anyway we will be executing them only if we are root)

Reply:
oh huh to be honest I have no idea why that's there it should not be

also the other commands here are important for debugging so we should not skip them if the user is not root, we should always execute them, if the current user is not sudo then they will be see that the script failed and thus they need to run it as root

though actually it would be great if the script mentions that it needs to be run as root/sudo user